### PR TITLE
Versioned docs

### PR DESCRIPTION
**Pull request by mitchdraft**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1221

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# If on fedora, run
+#     sudo dnf -y install gcc ruby-devel rubygems zlib-devel
+# to install html-proofer deps (only works with gcc, not clang!)
+install-tools:
+	gem install html-proofer
+
+.PHONY: docker-push
+docker-push: site
+	docker build -t gcr.io/solo-test-236622/gloo-docs:vabc6 .
```
Comment (rickducott) [inline docs/Makefile]: What is this?

Comment (mitchdraft) [reply docs/Makefile]: origin: https://github.com/solo-io/solo-docs/blob/master/gloo/Makefile#L36

currently a wip, soon will be a means of scoping docs by version

**Comment by mitchdraft** [comment]
todo:
- [ ] include change log docs generation in docs-push/build step
- [ ] restore the hand-written docs that used the `.sk.md` suffix (they get deleted by new clean step)
- [ ] cherry pick new additions to solo-docs
- [ ] trigger `gloo-docs:latest` image pulls on releases
- [ ] leverage solo-kit proto map (could do as a follow up)

**Comment by mitchdraft** [comment]
replaced by https://github.com/solo-io/gloo/pull/1335

**Pull request closed without merging**